main/src/main/java/cgeo/geocaching/connector/internal/InternalConnector.java

```diff
@@ -275,9 +270,22 @@ public static void assertHistoryCacheExists(final Context context) {
      * @return geocode      geocode of the newly created cache
      */
     public static String createCache(final Context context, @Nullable final String name, @Nullable final String description, final int assignedEmoji, @Nullable final Geopoint geopoint, final int listId) {
-        final long newId = DataStore.getNextAvailableInternalCacheId();
-        assertCacheExists(context, newId, name, description, assignedEmoji, geopoint, listId);
-        return geocodeFromId(newId);
```

Comment:
I believe method geocodeFromId() can be deleted too. This will also lead to a unit test in need to be removed